### PR TITLE
Generate default name for objects in carving. Closes: #1408

### PR DESCRIPTION
**Pull request by m-novikov**

### CODE REVIEW

File: ilastik/workflows/carving/carvingGui.py [hunk 1]
```diff
@@ -245,33 +246,46 @@ def onSegmentButton(self):
...
+
+    def getObjectNames(self):
+        return self.topLevelOperatorView.AllObjectNames[:].wait()
+
     def saveAsDialog(self, name=""):
         '''special functionality: reject names given to other objects'''
+        namesInUse = self.getObjectNames()
+
+        def generateObjectName():
+            return "{} {}".format(self.objectPrefix, len(namesInUse) + 1)
```
Comment (k-dominik) [inline ilastik/workflows/carving/carvingGui.py]: just a matter of taste, but I prefer the [F-string syntax](https://www.python.org/dev/peps/pep-0498/)...
```suggestion
            return f"{self.objectPrefix} {len(namesInUse) + 1}"
```

**Comment by m-novikov** [comment]
> Once I reach 10 objects, the sorting gets off (as it's alphanumerical) object 10 will be shown before object 2. That is also not really desirable.
I suppose we can add 0 padding then. If someone gets more than 99 elements with same prefix sorting will be minor problem.

**Comment by k-dominik** [comment]
> > Once I reach 10 objects, the sorting gets off (as it's alphanumerical) object 10 will be shown before object 2. That is also not really desirable.
> 
> I suppose we can add 0 padding then. If someone gets more than 99 elements with same prefix sorting will be minor problem.

Come to think of it, right now we do know that the number is added rightmost in the string. So padding would not really be necessary - however, I think it is more user-friendly to pad for readability with other tools. Do you think something with a `{placeholder}`, like we do in the export dialog, for the user to specify the number format could make sense?

**Comment by k-dominik** [comment]
This is very nice. I guess the cherry for the users would be to have an explanatory tooltip

**Pull request merged**